govuk-one-login/authentication-frontend
# BAU: Add a script to automatically build .env file for a given target environment

## What

Create a script which enables the auto-generation of a .env file for a provided target environment.

This will stop us from having to create, store and distribute multiple versions for each environment.

## How to review

- Code Review

1. remove your existing .env file (`mv .env .env.original`)
2. run the script `scripts/create-env-file build`. Observe that a new .env file has been created
3. run the frontend application `./startup.sh -lc`.
4. Access via the local stub (https://localhost:2000) Observe that there are no errors and you are redirected to the frontend
5. modify one of the variables above the 'DO NOT MODIFY AFTER THIS POINT' marker
6. run the script `scripts/create-env-file build`. Observe that your changed variable is still set
7. Add a new variable, rerun the script, observe that your new variable is still present under a 'unknown variables' header

8. Repeat the above steps for `sandpit, authdev#`
